### PR TITLE
Refactor shapes

### PR DESCRIPTION
**Pull request by mochen72**
Improved documentation and coding style of ShapeFunctions (and PlotOptions)

### CODE REVIEW

**Comment by sudo-michael** [comment]
I think we should follow normal python coding conversations. So all functions should start with a lowercase and use snake_case

File: odp/Shapes/ShapesFunctions.py [hunk 1]
```diff
@@ -90,7 +106,8 @@ def Upper_Half_Space(grid, dim, value):
             data += -grid.vs[i] + value
     return data
 
-def Union(shape1, shape2):
+
+def Union(shape1: np.ndarray, shape2: np.ndarray) -> np.ndarray:
```
Comment (sudo-michael) [inline odp/Shapes/ShapesFunctions.py:110]: Perhaps use a reduce function to support any number of args

https://numpy.org/doc/stable/reference/generated/numpy.ufunc.reduce.html

File: odp/Shapes/ShapesFunctions.py [hunk 2]
```diff
@@ -114,9 +132,36 @@ def Intersection(shape1, shape2):
...
+    """    
     data = np.zeros(grid.pts_each_dim)  
+
     for i in range(grid.dims):
         data = data + np.power(grid.vs[i] - center[i],2)/np.power(semiAxLen[i],2)
-    data = np.sqrt(data) - 1
-    return data
+
+    if quadratic:
+        return data - 1
```
Comment (sudo-michael) [inline odp/Shapes/ShapesFunctions.py:162]: Why is -1 necessary or maybe include the math formula in the documentation

**Comment by mochen72** [comment]
Good points. In general, how much renaming of functions should we do? Many people are already using this toolbox. Perhaps we can make the old names still usable.

What do you think? @sudo-michael @buinm

**Comment by sudo-michael** [comment]
> In general, how much renaming of functions should we do? Many

Perhaps it's out the scope of this PR, but I think a standard code style guide should be imposed on this project.

For example all the functions in https://github.com/SFU-MARS/optimized_dp/blob/master/odp/compute_trajectory.py
uses lower case + snake case.

> Many people are already using this toolbox. Perhaps we can make the old names still usable
We could add a `@depreciated` to the changed functions or a final refactor could be done before the eventual "1.0" release

**Comment by buinm** [comment]
Changes look good to me. I don't think renaming functions is worth the chances that other's codes stop working and require updates. Eventually the toolbox only has certain number of utitlity functions anyway.

**Pull request merged**